syncd/MdioIpcServer.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <ctype.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/select.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <time.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <pthread.h>
+
+extern "C" {
+#include <sai.h>
+#include <saiversion.h>
+}
+
+namespace syncd
+{
+    class MdioIpcServer
+    {
+        public:
+
+            MdioIpcServer(int globalContext);
+
+            virtual ~MdioIpcServer();
+
+        public:
+
+            static void setSwitchId(_In_ sai_object_id_t switchRid);
+
+            static void setSwitchMdioApi(sai_switch_api_t *switch_api);
```

Comment:
Each parameter is changed to be in a separate line. The missing _In_ is added. shared_ptr does not help here because  it won't protect from sai_api_uninitialize() and memset(&m_apis, 0, sizeof(m_apis)). A new function MdioIpcServer::clearSwitchMdioApi() is added to ensure the m_apis.switch_api won't be used after VendorSai::uninitialize().

Reply:
MdioIpcServer::setSwitchMdioApi does no longer exist.